server/sonar-server/src/main/java/org/sonar/server/qualitygate/changeevent/QGChangeEventListener.java

```diff
@@ -20,7 +20,40 @@
 package org.sonar.server.qualitygate.changeevent;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import org.sonar.api.rules.RuleType;
 
 public interface QGChangeEventListener {
-  void onChanges(Trigger trigger, Collection<QGChangeEvent> changeEvents);
+  /**
+   * @deprecated use {{@link #onIssueChanges(QGChangeEvent, Set)}} instead
+   */
+  // TODO remove this method and turn default method #onIssueChanges(QGChangeEvent, Set) into an interface method when
+  // support for #onIssueChanges(QGChangeEvent, Set) have been merged into sonar-branch's master
+  @Deprecated
+  default void onChanges(Trigger trigger, Collection<QGChangeEvent> changeEvents) {
+    // do nothing
+  }
+
+  /**
+   * Called consequently to a change done on one or more issue of a given project.
+   * 
+   * @param qualityGateEvent can not be {@code null}
+   * @param changedIssues can not be {@code null} nor empty
+   */
+  default void onIssueChanges(QGChangeEvent qualityGateEvent, Set<ChangedIssue> changedIssues) {
+    onChanges(Trigger.ISSUE_CHANGE, Collections.singleton(qualityGateEvent));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this use of "onChanges"; it is deprecated. ([squid:CallToDeprecatedMethod](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10085%2Fupgrade_qg_change_event_api_with_issue_details&issues=AWDCAF_iqDrCo5ysgzHA&open=AWDCAF_iqDrCo5ysgzHA)

[ISSUE_KEY]: AWDCAF_iqDrCo5ysgzHA